client/verta/tests/unit_tests/conftest.py

```diff
@@ -64,15 +71,137 @@ def __repr__(self):  # avoid network calls when displaying test results
 
 
 @pytest.fixture(scope="session")
-def mock_registered_model_version(mock_conn, mock_config):
-    """Return a mocked object of the RegisteredModelVersion class for use in tests"""
+def mock_simple_pipeline_definition() -> Dict[str, Any]:
+    """Return a mocked pipeline definition for use in tests"""
+
+    def simple_pipeline_definition(id: int) -> Dict[str, Any]:
+        return {
+            "graph": [
+                {"predecessors": [], "name": "step1"},
+                {"predecessors": ["step1"], "name": "step2"},
+            ],
+            "pipeline_version_id": id,
+            "steps": [
+                {
+                    "model_version_id": 1,
+                    "name": "step1",
+                },
+                {
+                    "model_version_id": 2,
+                    "name": "step2",
+                },
+            ],
+        }
+
+    return simple_pipeline_definition
+
+
+@pytest.fixture(scope="session")
+def make_mock_registered_model_version(
+    mock_conn, mock_config, mock_simple_pipeline_definition
+) -> Callable:
+    """Return a callable function for creating mocked objects of the
+    RegisteredModelVersion class for use in tests that require multiple
+    unique instances.
+    """
 
     class MockRegisteredModelVersion(RegisteredModelVersion):
         def __repr__(self):  # avoid network calls when displaying test results
             return object.__repr__(self)
 
-    return MockRegisteredModelVersion(
-        mock_conn,
-        mock_config,
-        _RegistryService.ModelVersion(id=555, registered_model_id=123),
-    )
+        def _get_artifact(self, key=None, artifact_type=None):
+            return json.dumps(mock_simple_pipeline_definition(id=self.id)).encode(
+                "utf-8"
+            )
+
+    def _make_mock_registered_model_version():
+        """Return a mocked ``RegisteredModelVersion``.
+
+        ``id`` and ``registered_model_id`` will be random and unique for the
+        test session.
+
+        """
+        model_ver_id = random.randint(1, 1000000)
+        reg_model_id = random.randint(1, 1000000)
+
+        return MockRegisteredModelVersion(
+            mock_conn,
+            mock_config,
+            _RegistryService.ModelVersion(
+                id=model_ver_id,
+                registered_model_id=reg_model_id,
+                version="test_version_name",
+            ),
+        )
+
+    return _make_mock_registered_model_version
+
+
+@pytest.fixture(scope="session")
+def make_mock_pipeline_step(make_mock_registered_model_version) -> Callable:
+    """
+    Return a callable function for creating mocked objects of the PipelineStep
+    class for use in tests that require multiple unique instances.
+    """
+
+    class MockPipelineStep(PipelineStep):
+        def __repr__(self):
```

Comment:
We should keep this comment, otherwise it's not very clear what this is for.

```suggestion
        def __repr__(self):  # avoid network calls when displaying test results
```